fix: FIT-1097: Viewing Data Manager under Annotators account show disabled state of 'Task State' column; I can't use neither Columns nor Filters

yyassi-heartex opened the pull request:
This pull request introduces improvements to how state transition information is displayed and serialized, focusing on user privacy and a clearer UI. The most notable changes are the conditional exclusion of the `triggered_by` field for annotators in API responses, and a refactor of the state history popover component to streamline how transition reasons are shown.

Annotators now have access to Task State
<img width="408" height="500" alt="image" src="https://github.com/user-attachments/assets/7a501e82-ce1b-4bf7-b4de-8b3ffe394b12" />
<img width="582" height="457" alt="image" src="https://github.com/user-attachments/assets/3a2ce1c7-2343-4c80-ab67-da0fe8785ebc" />


**Backend serialization and privacy:**

* Updated the `StateModelSerializer` in `label_studio/fsm/serializers.py` to exclude the `triggered_by` field from API responses when the requesting user is an annotator, enhancing privacy for certain user roles.

**Frontend UI improvements:**

* Refactored the `StateHistoryPopoverContent` component in `state-history-popover-content.tsx` to consistently display either the user who triggered a state change or the contextual reason, using a single variable for clarity and improved maintainability.

yyassi-heartex committed:
fix: FIT-1097: Viewing Data Manager under Annotators account show disabled state of 'Task State' column; I can't use neither Columns nor Filters

bmartel commented on label_studio/fsm/serializers.py:
```diff
@@ -23,6 +23,20 @@ class StateModelSerializer(serializers.Serializer):
+        if request and hasattr(request, 'user'):
+            user = request.user
+            if getattr(user, 'is_annotator', False):
+                data.pop('triggered_by', None)
```
This says it doesn't include the `triggered_by` for annotators or reviewers, but checks `is_annotator`  only.

yyassi-heartex replied:
Yeah I re-enabled it for reviewers since I see they have access to the annotator column which has info about that user type. I didnt update the comment. I can either put the check for reviewer back or update the comment

yyassi-heartex replied:
Yeah I re-enabled it for reviewers since I see they have access to the annotator column which has info about that user type. I didnt update the comment. I can either put the check for reviewer back or update the comment

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-fit-1097/state-column-access

yyassi-heartex committed:
cleaning up comment

bmartel commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/20032399860)

bmartel committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20032399860

bmartel committed:
Merge branch 'develop' into 'fb-fit-1097/state-column-access'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20032399860
